Add unit tests for conversation CRUD handlers

igorveselinovic opened the pull request:
Resolves #1, #2 

Making a PR for just the conversation CRUD handlers unit tests to get some feedback before I write the tests for the user-to-conversation mappings handlers. Used the mock database methods to track what is being passed as arguments, so that the unit tests can validate that the methods are being called correctly.

Added a Dockerfile and Makefile and the GitHub actions CI workflow config. Mostly the same as what's in `pest-control`, with a few small differences.

igorveselinovic committed:
Add unit tests for conversation CRUD handlers

Added a mock DB struct and helper functions for creating pointers.

igorveselinovic committed:
Add Dockerfile and Makefile

Also make logging message for initial DB opening more clear.

igorveselinovic committed:
Add CI workflow for building and unit testing

igorveselinovic committed:
Clean up CI yaml and use more descriptive names

igorveselinovic committed:
Shorten CI job name

Thao-Tran commented on models/mock.go:
```diff
@@ -0,0 +1,60 @@
+	Conversation *Conversation
+	Mapping      *UserConversationMapping
+	Mappings     []*UserConversationMapping
+	Error        error
```
Is there a case where there's multiple DB methods being invoked in a handler and you would want to throw an error for one but not the others?

igorveselinovic replied:
Yeah, there probably will be. I don't think I actually even use this field at all yet. I'll just make it a list then, I guess, so it could be something like:
```
[nil, nil, nil, err]
```
if I want the 4th call to any DB method to return a specific error. Does that makes sense?

Thao-Tran replied:
Yeah I think that makes sense and there can be a counter in the struct to keep track of the calls.

igorveselinovic committed:
Make the mock DB have a list of errors

If it is nil, then no error will be returned. If it has stuff in it,
then it will use a count that is incremented each time a mock method is
called to index the error list.

mRileyMacK commented on models/mock.go:
```diff
@@ -0,0 +1,88 @@
+	if db.Errors == nil {
+		return nil
+	} else {
+		err := db.Errors[db.Count]
```
I concur

igorveselinovic replied:
Well, when you put it that way...

Thao-Tran commented on models/mock.go:
```diff
@@ -4,57 +4,85 @@ type MockDB struct {
+	if db.Errors == nil {
+		return nil
+	} else {
+		err := db.Errors[db.Count]
```
Should probably check that `db.Count` doesn't exceed the size of `db.Errors`

igorveselinovic replied:
I disagree

igorveselinovic committed:
Add check for mock count exceeding error list size
